Firmware/first_lay_cal.cpp

```diff
@@ -37,12 +37,30 @@ static constexpr float spacing(float layer_height, float extrusion_width, float
 static void lay1cal_common_enqueue_loop(const char * const * cmd_sequence, const uint8_t steps) {
     for (uint8_t i = 0; i < steps; ++i)
     {
-        enquecommand_P(static_cast<char*>(pgm_read_ptr(cmd_sequence + i)));
+        void * pgm_ptr = pgm_read_ptr(cmd_sequence + i);
+
+        // M702 is currently only used with MMU enabled
+        if (pgm_ptr == MSG_M702 && !MMU2::mmu2.Enabled()) {
+            continue;
+        }
+
+        enquecommand_P(static_cast<char*>(pgm_ptr));
     }
 }
 
-static const char extrude_fmt[] PROGMEM = "G1 X%d Y%d E%-.5f";
-static const char zero_extrusion[] PROGMEM = "G92 E0";
+static const char extrude_fmt_X[] PROGMEM = "G1X%gE%g";
```

Comment:
nope. If we end up scientific notation then this will fail for sure. Maybe just a simple %f might be better suited. That doesn't pad the end with zeros, no?

Reply:
I don't think that the avr will produce any `e+2` and also never seen anything longer than 7 chars `"G1Y23.6111E0.74031"`